numba/tests/test_array_manipulation.py

```diff
@@ -585,6 +589,132 @@ def test_bad_float_index_npm(self):
         self.assertIn('unsupported array index type float64',
                       str(raises.exception))
 
+    def test_fill_diagonal(self):
+        self._fill_diagonal_basic()
+        self._fill_diagonal_exception_cases()
```

Comment:
Perhaps move the exception cases out to another test so the disabled leak checker only applies to that test?

Reply:
Sure, makes sense.